### PR TITLE
Fix https://github.com/arc298/instagram-scraper/issues/805

### PR DESCRIPTION
**Pull request by nns33213**
Endpoints used by instagram-scraper et al. for getting user information and media info no longer work. New method uses Instagram API calls, consistent with the current version of the web app.

Fixes:
https://github.com/arc298/instagram-scraper/issues/805
https://github.com/arc298/instagram-scraper/issues/806
https://github.com/arc298/instagram-scraper/issues/807

### CODE REVIEW

**Comment by jasonkhanlar** [comment]
Is it okay to push this update so that `pip install instagram-scraper` includes it?

**Comment by nns33213** [comment]
> Is it okay to push this update so that `pip install instagram-scraper --update` includes it?

@jasonkhanlar maintainer of this project is currently AWOL ;)

**Comment by jasonkhanlar** [comment]
> > Is it okay to push this update so that `pip install instagram-scraper --update` includes it?
> 
> @jasonkhanlar maintainer of this project is currently AWOL ;)

Oh, bummer! Since this project uses Unlicense license, are any non-AWOL devs okay with forking the project to be maintained with an alternatively named package at https://pypi.org/ ?

**Comment by nns33213** [comment]
> Oh, bummer! Since this project uses Unlicense license, are any non-AWOL devs okay with forking the project to be maintained with an alternatively named package at https://pypi.org/ ?

@jasonkhanlar I mean, he'll probably be back in a couple of days to release the new version to pip.

**Comment by AlexNik** [comment]
I had a vacation for the last week without the Internet.
@nns33213 does this PR fix the all new recent problems?
Which problems are still alive?

**Comment by nns33213** [comment]
@AlexNik this fixes the main problem that Instagram changed the APIs. The current release on PIP is unusable.

Is there a reason we do
https://github.com/arc298/instagram-scraper/blob/58b92b00a72e31805c6c4f953904fc23e9a0c253/instagram_scraper/app.py#L30
rather than `import .constants` or `import constants`? This messes up the development because it imports from the pip module and not the version you have in the current folder.

Also why do we login with iPhone user agent and then switch to Chrome? Is that still necessary? The code that introduced that is from 2018.

File: instagram_scraper/app.py [hunk 1]
```diff
@@ -882,27 +873,12 @@ def get_media(self, dst, executor, future_to_item, user):
...
-                            userinfo = self.deep_get(self._get_json(shared_data), 'graphql.user')
-            except (TypeError, KeyError, IndexError):
-                pass
+        try:
+            resp = self.get_json(WEB_PROFILE_INFO.format(username), headers=API_HEADERS)
 
-        return userinfo
+            return self._get_json(resp)['data']['user']
+        except:
+            return None
```
[the remark this answers is not in the document]
Comment (AlexNik) [reply instagram_scraper/app.py:881]: @nns33213 Please add logging here and I merge it

**Comment by AlexNik** [comment]
> Is there a reason we do
> rather than `from .constants` or `from constants`? This messes up the development because it imports from the pip module and not the version you have in the current folder.
Need to be fixed.

> Also why do we login with iPhone user agent and then switch to Chrome? Is that still necessary? The code that introduced that is from 2018.
I don't know)))
Here is some magic with user agents. Unfortunatelly the code of this project is very old and here is a lot of patterns for Python 2.7... I want to refactor it but it is very difficult and laborious. It is my dream but I don't have a lot of time. The structure of Instaloader project is better 😞

**Comment by nns33213** [comment]
@victoryy2003 I fixed your `--profile-metadata` problem.

@AlexNik really please merge ASAP, the app hasn't been working for nearly 2 weeks at this point.

https://github.com/arc298/instagram-scraper/issues/815
https://github.com/arc298/instagram-scraper/issues/816
https://github.com/arc298/instagram-scraper/issues/817

**Comment by djmarioka** [comment]
Hi, do you know when is going the app to work again? 

Thanks for the efforts to the developers.

**Comment by akhepcat** [comment]
Not sure if this is a result of other changes, or the refactor to work after this lastest fix,  but  with  nns33213's fixes, I'm unable to scrape "private" accounts, even while providing the login data.

**Comment by nns33213** [comment]
@akhepcat try https://github.com/arc298/instagram-scraper/pull/809/commits/5b77b7e1a62b541f7e5f0c65e1addd9487d32643

**Comment by akhepcat** [comment]
> @akhepcat try [5b77b7e](https://github.com/arc298/instagram-scraper/commit/5b77b7e1a62b541f7e5f0c65e1addd9487d32643)

That seems to have done it!

Thanks for the quick fix.

**Review by AlexNik — APPROVED** [review]

**Comment by djmarioka** [comment]
Hi, 

I've updated the newest version, and it continues triggering the **message: "Error getting user details for xxxx. Please verify that the user exists**", I attach a screenshot:

ERROR INSTAGRAM SCRAPER

Could you fix it? 

Thanks.

**Comment by nns33213** [comment]
> Hi,
> 
> I've updated the newest version, and it continues triggering the **message: "Error getting user details for xxxx. Please verify that the user exists**", I attach a screenshot:
> 
> ERROR INSTAGRAM SCRAPER
> 
> Could you fix it?
> 
> Thanks.

@djmarioka are you sure that you actually updated the scraper correctly using `pip`? This error mode basically doesn't exist anymore, even if your account is blocked. It worries me that you have a folder open called `INSTAGRAM SCRAPER 17.06.2022` but you're not calling any file in that folder.

@AlexNik thanks for merging!

**Comment by djmarioka** [comment]
Hi, 

sorry, it was my fault.

I was executing it through a proxy, now it works again. :)

Thanks.

**Pull request merged**